k4MarlinWrapper/src/components/EDM4hep2Lcio.cpp

```diff
@@ -340,8 +393,32 @@ void EDM4hep2LcioTool::convertAdd(const std::string& e4h_coll_name, const std::s
   } else if (fulltype == "edm4hep::EventHeader") {
     convertEventHeader(e4h_coll_name, lcio_event);
   } else if (fulltype == "edm4hep::ParticleID") {
-    pidCollections.emplace_back(e4h_coll_name, static_cast<const edm4hep::ParticleIDCollection*>(collPtr),
-                                edm4hep::utils::PIDHandler::getAlgoInfo(metadata, e4h_coll_name));
+    std::optional<std::string>              maybeAlgoName;
+    std::optional<int>                      maybeAlgoType;
+    std::optional<std::vector<std::string>> maybeParamNames;
+
+    if (m_podioDataSvc) {
+      const auto& frame = metadata.value().get();
+      maybeAlgoName =
+          frame.getParameter<std::string>(podio::collMetadataParamName(e4h_coll_name, edm4hep::labels::PIDAlgoName));
+      maybeAlgoType =
+          frame.getParameter<int>(podio::collMetadataParamName(e4h_coll_name, edm4hep::labels::PIDAlgoType));
+      maybeParamNames = frame.getParameter<std::vector<std::string>>(
+          podio::collMetadataParamName(e4h_coll_name, edm4hep::labels::PIDParameterNames));
+
+    } else {
+      maybeAlgoName = k4FWCore::getParameter<std::string>(
+          podio::collMetadataParamName(e4h_coll_name, edm4hep::labels::PIDAlgoName));
+      maybeAlgoType =
+          k4FWCore::getParameter<int>(podio::collMetadataParamName(e4h_coll_name, edm4hep::labels::PIDAlgoType));
+      maybeParamNames = k4FWCore::getParameter<std::vector<std::string>>(
+          podio::collMetadataParamName(e4h_coll_name, edm4hep::labels::PIDParameterNames));
+    }
+
+    edm4hep::utils::ParticleIDMeta pidInfo{std::move(maybeAlgoName.value()), maybeAlgoType.value(),
+                                           maybeParamNames.value()};
```

Comment:
This should also be covered by https://github.com/key4hep/k4FWCore/pull/273, I think.

Reply:
It can but then when using the `PodioDataSvc` the `pidInfo` has to be created like it is shown here so all the labels and `podio::collMetadataParamName` have to be used anyway, see a few lines above. I'm not sure what's better, at least now they are done exactly the same way when using `PodioDataSvc` and when not. Using https://github.com/key4hep/k4FWCore/pull/273 could mean the implementations could differ at some point in time...

Reply:
Done